src/test/java/org/broadinstitute/hellbender/utils/nio/NioTests.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.gcloud.AuthCredentials;
+import com.google.gcloud.RetryParams;
+import com.google.gcloud.storage.StorageOptions;
+import com.google.gcloud.storage.contrib.nio.CloudStorageConfiguration;
+import com.google.gcloud.storage.contrib.nio.CloudStorageFileSystem;
+
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Test GCS access via the NIO APIs.
+ */
+public final class NioTests extends BaseTest {
+
+  @Test
+  public void testGcsEnabled() {
+    FileSystem fs = FileSystems.getFileSystem(URI.create("gs://domain-registry-alpha"));
+  }
+
+  @Test
+  public void openPublicFile() throws IOException {
+    InputStream inputStream = Files.newInputStream(Paths.get(URI.create("gs://pgp-harvard-data-public/hu011C57/GS000018120-DID/GS000015172-ASM/manifest.all.sig")));
```

Comment:
I wonder if hadoop's Path has the same distinction.  That could explain some of our weird bugs.  I should check that since we use strings and paths and uri's pretty interchangeably for a lot of things.  If the different getters have different meanings that could be problematic in our existing code..